packages/blockchain-link/src/workers/solana/index.ts

```diff
@@ -280,17 +278,28 @@ const getAccountInfo = async (request: Request<MessageTypes.GetAccountInfo>) =>
             .filter((tx): tx is Transaction => !!tx);
     };
 
-    const tokenAccounts = await api.rpc
-        .getTokenAccountsByOwner(
-            publicKey,
-            { programId: address(TOKEN_PROGRAM_PUBLIC_KEY) } /* filter */,
-            {
-                encoding: 'jsonParsed',
-            },
+    const getTokenAccounts = (tokenProgramId: string) =>
+        api.rpc
+            .getTokenAccountsByOwner(
+                publicKey,
+                { programId: address(tokenProgramId) } /* filter */,
+                {
+                    encoding: 'jsonParsed',
+                },
+            )
+            .send();
+
+    const allTokenAccounts = (
+        await Promise.all(
+            Object.values(tokenPrograms).map(programInfo =>
```

Comment:
nit: since the variable in `map` is called `programInfo` I'd also consider renaming the `tokenPrograms` constant to `tokenProgramsInfo`.